src/cloudant/index.py

```diff
@@ -262,3 +262,91 @@ def delete(self):
         """
         msg = 'Deleting the \"special\" index is not allowed.'
         raise CloudantException(msg)
+
+class SearchIndex(dict):
+    """
+    Encapsulates a SearchIndex as a dictionary based object, exposing the
+    search index function and analyzer as attributes.
+    """
```

Comment:
I assume you will be adding more content here to mirror that of the [View](https://github.com/cloudant/python-cloudant/blob/master/src/cloudant/view.py#L28-L92) docstring?